src/coreclr/jit/simdashwintrinsic.cpp

```diff
@@ -405,6 +405,22 @@ GenTree* Compiler::impSimdAsHWIntrinsicSpecial(NamedIntrinsic       intrinsic,
         }
 #endif // TARGET_X86
 
+#if defined(TARGET_XARCH)
```

Comment:
The issue is that this is actually under a `#if defined(TARGET_XARCH)` already (defined on L385) and so this is never hit on x86.

The simplest fix is to remove the `#if defined(TARGET_XARCH)` here and then copy the following logic into the `#elif defined(TARGET_ARM64)` on L446
```cpp
    switch (intrinsic)
    {
        case NI_VectorT128_As:
        {
            unsigned  retSimdSize;
            var_types retBaseType = getBaseTypeAndSizeOfSIMDType(sig->retTypeSigClass, &retSimdSize);

            if (!varTypeIsArithmetic(retBaseType) || (retSimdSize == 0))
            {
                // We get here if the return type is an unsupported type
                return nullptr;
            }
            break;
        }

        default:
        {
            // Most intrinsics have some path that works even if only AdvSimd is available
            break;
        }
    }
```

Reply:
Got it. I'll look at the structure and find the opportunity to share code.

Reply:
I reordered the ifdef block into two. One for general instruction set availability, one for special logic for each instruction. Is this sure?

Reply:
It looks correct. I think this will be good to merge if CI passes